Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -594,6 +598,22 @@ final class BuildOperationBuildSystemDelegateHandler: LLBuildBuildSystemDelegate
     func shouldResolveCycle(rules: [BuildKey], candidate: BuildKey, action: CycleAction) -> Bool {
         return false
     }
+    
+    func preparationStepStarted(_ name: String) {
```

Comment:
Not in love with this but I think this is conceptually correct as long as these steps happen outside the build system — I tried impersonating build commands, but it made things harder to understand, and so this is at least clear.  We should consider refactoring the output emission here and should revisit this at that time.  For now this seems the most correct.

Reply:
(this comment applies to these three methods, calling out build preparation steps as distinct from the build — in fact I think there are other steps that could fall into this category, to the extent SwiftPM generates modulemaps or does other things that could take time and/or fail)